core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/JDBCUserStoreManager.java

```diff
@@ -2096,8 +2096,14 @@ public void doSetUserClaimValues(String userName, Map<String, String> claims, St
             while (ite2.hasNext()) {
                 Map.Entry<String, String> entry = ite2.next();
                 String claimURI = entry.getKey();
-                if (availableProperties.containsKey(claimPropertyMap.get(claimURI))) {
-                    availableClaims.put(claimURI, entry.getValue());
+                String claimValue = claimPropertyMap.get(claimURI);
+                if (availableProperties.containsKey(claimValue)) {
```

Comment:
containsKey() might throw NPE if the key is null. In this case, as availableProperties is a HashMap, it won't through the NPE. But it's always better to program to the interface. Hence, Please add the null check.